roles/oas-pre-run/tasks/main.yml

```diff
@@ -0,0 +1,22 @@
+---
+- name: import centos gpg keys
+  rpm_key:
+    state: present
+    key: "{{ os_gpg_key.centos.key_url }}"
+  when: ansible_distribution in ['CentOS']
+
+- name: import redhat gpg keys
+  rpm_key:
+    state: present
+    key: "{{ os_gpg_key.redhat.key_url }}"
+  when: ansible_distribution in ['RedHat']
+
+- name: check the password hashing alogrithm
+  shell: authconfig --test | grep sha512
+  register: result
+  ignore_errors: true
+  when: ansible_distribution in ['CentOS', 'RedHat']
```

Comment:
you can use os_ansible_family instead of ansible_distribution in ['CentOS', 'RedHat']